rotkehlchen/tests/utils/blockchain.py

```diff
@@ -627,28 +633,18 @@ def mock_avax_balance(account):
             return FVal(1)
         return ZERO
 
-    def mock_optimism_has_activity(account):
-        if account in optimism_addresses:
+    def mock_chain_has_activity(account: ChecksumEvmAddress, chain: SUPPORTED_EVM_CHAINS):
+        if chain == SupportedBlockchain.ETHEREUM and account in ethereum_addresses:
```

Comment:
You tried to abstract this here but failed to do so since you just moved the different if checks into one huge if/elif/elif here.

Just be smarter. This is a test and you can do stuff that you should not do in production code but can do here: `addresses = locals()[f'{chain.get_key()}_addresses']`

I guess you can understand what this means.  The function gets reduced to:

```python
addresses = locals().get(f'{chain.get_key()}_addresses')
return EtherscanHasChainActivity.TRANSACTIONS if addresses is not None and account in addresses else EtherscanHasChainActivity.NONE  # noqa: E501
```

Now if this was production we would have done something similar but in a probably less dirty way. As stuff like `locals`, `getattr` and `setattr` is dynamic programming which is ugly and hard to maintain/debug and all typing/linting toosl can't check them.

Reply:
Ok, I did it a bit different, because locals() in the inside function doesn't have access to the outer scope variables. So I constructed a dictionary and access its keys.

Reply:
you can also just use `globals` if the stuff you need are outside the function. I need to see if what you did makes sense.

Reply:
As I saw, globals too will probably not do it since it does not have access to the outer function's local variables. This is why I used the dictionary approach.